From a9de4f93d52a9338233177a4f6f3a2f165f7f071 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Wed, 24 Apr 2024 15:50:58 +0200
Subject: [PATCH 1120/1141] arm64: dts: st: fix #dma-cells and dmas properties
 on stm32mp25

DMA3 bindings define
  "#dma-cells":
    const: 3

So hpdma nodes must have #dma-cells = <3>;

stm32-dma3 driver only takes care of the first 3 parameters of dmas
property. Any extra parameter is ignored.
Moreover, the fourth parameter removed, is always 0. So it was also unused
by DMA clients.
Get rid of this extra parameter, to avoid any confusion and misalignment
with stm32-dma3 bindings.

Depends-on: Icae39e7521c06373e9158353835ae689eafe2be6
Change-Id: I7c25e7085daacb12f719c1e0e4657b978dc6840e
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/376315
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi  | 182 ++++++++++++------------
 arch/arm64/boot/dts/st/stm32mp25xc.dtsi |   8 +-
 arch/arm64/boot/dts/st/stm32mp25xf.dtsi |   8 +-
 3 files changed, 99 insertions(+), 99 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -338,7 +338,7 @@
 				     <GIC_SPI 48 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&scmi_clk CK_SCMI_HPDMA1>;
 			power-domains = <&RET_PD>;
-			#dma-cells = <4>;
+			#dma-cells = <3>;
 			st,axi-max-burst-len = <16>;
 		};
 
@@ -363,7 +363,7 @@
 				     <GIC_SPI 64 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&scmi_clk CK_SCMI_HPDMA2>;
 			power-domains = <&RET_PD>;
-			#dma-cells = <4>;
+			#dma-cells = <3>;
 			st,axi-max-burst-len = <16>;
 		};
 
@@ -388,7 +388,7 @@
 				     <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&scmi_clk CK_SCMI_HPDMA3>;
 			power-domains = <&RET_PD>;
-			#dma-cells = <4>;
+			#dma-cells = <3>;
 			st,axi-max-burst-len = <16>;
 		};
 
@@ -423,8 +423,8 @@
 				compatible = "st,stm32mp25-omi";
 				reg = <0 0 0x400>;
 				interrupts = <GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>;
-				dmas = <&hpdma 2 0x62 0x00003121 0x0>,
-				       <&hpdma 2 0x42 0x00003112 0x0>;
+				dmas = <&hpdma 2 0x62 0x00003121>,
+				       <&hpdma 2 0x42 0x00003112>;
 				dma-names = "tx", "rx";
 				st,syscfg-dlyb = <&syscfg 0x1000>;
 				clocks = <&scmi_clk CK_SCMI_OSPI1>;
@@ -438,8 +438,8 @@
 				compatible = "st,stm32mp25-omi";
 				reg = <1 0 0x400>;
 				interrupts = <GIC_SPI 164 IRQ_TYPE_LEVEL_HIGH>;
-				dmas = <&hpdma 3 0x62 0x00003121 0x0>,
-				       <&hpdma 3 0x42 0x00003112 0x0>;
+				dmas = <&hpdma 3 0x62 0x00003121>,
+				       <&hpdma 3 0x42 0x00003112>;
 				dma-names = "tx", "rx";
 				st,syscfg-dlyb = <&syscfg 0x1400>;
 				clocks = <&scmi_clk CK_SCMI_OSPI2>;
@@ -806,8 +806,8 @@
 				clocks = <&rcc CK_BUS_SPI2>, <&rcc CK_KER_SPI2>;
 				clock-names = "pclk", "i2sclk";
 				resets = <&rcc SPI2_R>;
-				dmas = <&hpdma 51 0x43 0x12 0>,
-					<&hpdma 52 0x43 0x21 0>;
+				dmas = <&hpdma 51 0x43 0x12>,
+					<&hpdma 52 0x43 0x21>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI2_ID>;
 				power-domains = <&RET_PD>;
@@ -822,8 +822,8 @@
 				interrupts = <GIC_SPI 113 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_SPI2>;
 				resets = <&rcc SPI2_R>;
-				dmas = <&hpdma 51 0x20 0x00003012 0>,
-				       <&hpdma 52 0x20 0x00003021 0>;
+				dmas = <&hpdma 51 0x20 0x00003012>,
+				       <&hpdma 52 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI2_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -838,8 +838,8 @@
 				clocks = <&rcc CK_BUS_SPI3>, <&rcc CK_KER_SPI3>;
 				clock-names = "pclk", "i2sclk";
 				resets = <&rcc SPI3_R>;
-				dmas = <&hpdma 53 0x43 0x12 0>,
-					<&hpdma 54 0x43 0x21 0>;
+				dmas = <&hpdma 53 0x43 0x12>,
+					<&hpdma 54 0x43 0x21>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI3_ID>;
 				power-domains = <&RET_PD>;
@@ -854,8 +854,8 @@
 				interrupts = <GIC_SPI 125 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_SPI3>;
 				resets = <&rcc SPI3_R>;
-				dmas = <&hpdma 53 0x20 0x00003012 0>,
-				       <&hpdma 54 0x20 0x00003021 0>;
+				dmas = <&hpdma 53 0x20 0x00003012>,
+				       <&hpdma 54 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI3_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -869,8 +869,8 @@
 				clocks = <&rcc CK_KER_SPDIFRX>;
 				clock-names = "kclk";
 				interrupts = <GIC_SPI 170 IRQ_TYPE_LEVEL_HIGH>;
-				dmas = <&hpdma 71 0x43 0x212 0>,
-					<&hpdma 72 0x43 0x212 0>;
+				dmas = <&hpdma 71 0x43 0x212>,
+					<&hpdma 72 0x43 0x212>;
 				dma-names = "rx", "rx-ctrl";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPDIFRX_ID>;
 				status = "disabled";
@@ -881,8 +881,8 @@
 				reg = <0x400e0000 0x400>;
 				interrupts-extended = <&exti1 27 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART2>;
-				dmas = <&hpdma 11 0x20 0x10012 0x0>,
-				       <&hpdma 12 0x20 0x3021 0x0>;
+				dmas = <&hpdma 11 0x20 0x10012>,
+				       <&hpdma 12 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_USART2_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -895,8 +895,8 @@
 				reg = <0x400f0000 0x400>;
 				interrupts-extended = <&exti1 28 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART3>;
-				dmas = <&hpdma 13 0x20 0x10012 0x0>,
-				       <&hpdma 14 0x20 0x3021 0x0>;
+				dmas = <&hpdma 13 0x20 0x10012>,
+				       <&hpdma 14 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_USART3_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -909,8 +909,8 @@
 				reg = <0x40100000 0x400>;
 				interrupts-extended = <&exti1 30 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART4>;
-				dmas = <&hpdma 15 0x20 0x10012 0x0>,
-				       <&hpdma 16 0x20 0x3021 0x0>;
+				dmas = <&hpdma 15 0x20 0x10012>,
+				       <&hpdma 16 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART4_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -923,8 +923,8 @@
 				reg = <0x40110000 0x400>;
 				interrupts-extended = <&exti1 31 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART5>;
-				dmas = <&hpdma 17 0x20 0x10012 0x0>,
-				       <&hpdma 18 0x20 0x3021 0x0>;
+				dmas = <&hpdma 17 0x20 0x10012>,
+				       <&hpdma 18 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART5_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -941,8 +941,8 @@
 				resets = <&rcc I2C1_R>;
 				#address-cells = <1>;
 				#size-cells = <0>;
-				dmas = <&hpdma 27 0x20 0x00003012 0>,
-				       <&hpdma 28 0x20 0x00003021 0>;
+				dmas = <&hpdma 27 0x20 0x00003012>,
+				       <&hpdma 28 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_I2C1_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -958,8 +958,8 @@
 				resets = <&rcc I2C2_R>;
 				#address-cells = <1>;
 				#size-cells = <0>;
-				dmas = <&hpdma 30 0x20 0x00003012 0>,
-				       <&hpdma 31 0x20 0x00003021 0>;
+				dmas = <&hpdma 30 0x20 0x00003012>,
+				       <&hpdma 31 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_I2C2_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -975,8 +975,8 @@
 				resets = <&rcc I2C3_R>;
 				#address-cells = <1>;
 				#size-cells = <0>;
-				dmas = <&hpdma 33 0x20 0x00003012 0>,
-				       <&hpdma 34 0x20 0x00003021 0>;
+				dmas = <&hpdma 33 0x20 0x00003012>,
+				       <&hpdma 34 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_I2C3_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -992,8 +992,8 @@
 				resets = <&rcc I2C4_R>;
 				#address-cells = <1>;
 				#size-cells = <0>;
-				dmas = <&hpdma 36 0x20 0x00003012 0>,
-				       <&hpdma 37 0x20 0x00003021 0>;
+				dmas = <&hpdma 36 0x20 0x00003012>,
+				       <&hpdma 37 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_I2C4_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1009,8 +1009,8 @@
 				resets = <&rcc I2C5_R>;
 				#address-cells = <1>;
 				#size-cells = <0>;
-				dmas = <&hpdma 39 0x20 0x00003012 0>,
-				       <&hpdma 40 0x20 0x00003021 0>;
+				dmas = <&hpdma 39 0x20 0x00003012>,
+				       <&hpdma 40 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_I2C5_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1026,8 +1026,8 @@
 				resets = <&rcc I2C6_R>;
 				#address-cells = <1>;
 				#size-cells = <0>;
-				dmas = <&hpdma 42 0x20 0x00003012 0>,
-				       <&hpdma 43 0x20 0x00003021 0>;
+				dmas = <&hpdma 42 0x20 0x00003012>,
+				       <&hpdma 43 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_I2C6_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1043,8 +1043,8 @@
 				resets = <&rcc I2C7_R>;
 				#address-cells = <1>;
 				#size-cells = <0>;
-				dmas = <&hpdma 45 0x20 0x00003012 0>,
-				       <&hpdma 46 0x20 0x00003021 0>;
+				dmas = <&hpdma 45 0x20 0x00003012>,
+				       <&hpdma 46 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_I2C7_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1225,8 +1225,8 @@
 				reg = <0x40220000 0x400>;
 				interrupts-extended = <&exti1 29 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART6>;
-				dmas = <&hpdma 19 0x20 0x10012 0x0>,
-				       <&hpdma 20 0x20 0x3021 0x0>;
+				dmas = <&hpdma 19 0x20 0x10012>,
+				       <&hpdma 20 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_USART6_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1242,8 +1242,8 @@
 				clocks = <&rcc CK_BUS_SPI1>, <&rcc CK_KER_SPI1>;
 				clock-names = "pclk", "i2sclk";
 				resets = <&rcc SPI1_R>;
-				dmas = <&hpdma 49 0x43 0x12 0>,
-					<&hpdma 50 0x43 0x21 0>;
+				dmas = <&hpdma 49 0x43 0x12>,
+					<&hpdma 50 0x43 0x21>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI1_ID>;
 				power-domains = <&RET_PD>;
@@ -1258,8 +1258,8 @@
 				interrupts = <GIC_SPI 112 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_SPI1>;
 				resets = <&rcc SPI1_R>;
-				dmas = <&hpdma 49 0x20 0x00003012 0>,
-				       <&hpdma 50 0x20 0x00003021 0>;
+				dmas = <&hpdma 49 0x20 0x00003012>,
+				       <&hpdma 50 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI1_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1274,8 +1274,8 @@
 				interrupts = <GIC_SPI 152 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_SPI4>;
 				resets = <&rcc SPI4_R>;
-				dmas = <&hpdma 55 0x20 0x00003012 0>,
-				       <&hpdma 56 0x20 0x00003021 0>;
+				dmas = <&hpdma 55 0x20 0x00003012>,
+				       <&hpdma 56 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI4_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1383,8 +1383,8 @@
 				interrupts = <GIC_SPI 153 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_SPI5>;
 				resets = <&rcc SPI5_R>;
-				dmas = <&hpdma 57 0x20 0x00003012 0>,
-				       <&hpdma 58 0x20 0x00003021 0>;
+				dmas = <&hpdma 57 0x20 0x00003012>,
+				       <&hpdma 58 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI5_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1411,7 +1411,7 @@
 					#sound-dai-cells = <0>;
 					clocks = <&rcc CK_KER_SAI1>;
 					clock-names = "sai_ck";
-					dmas = <&hpdma 73 0x43 0x21 0>;
+					dmas = <&hpdma 73 0x43 0x21>;
 					power-domains = <&CLUSTER_PD>;
 					status = "disabled";
 				};
@@ -1422,7 +1422,7 @@
 					#sound-dai-cells = <0>;
 					clocks = <&rcc CK_KER_SAI1>;
 					clock-names = "sai_ck";
-					dmas = <&hpdma 74 0x43 0x12 0>;
+					dmas = <&hpdma 74 0x43 0x12>;
 					power-domains = <&CLUSTER_PD>;
 					status = "disabled";
 				};
@@ -1448,7 +1448,7 @@
 					#sound-dai-cells = <0>;
 					clocks = <&rcc CK_KER_SAI2>;
 					clock-names = "sai_ck";
-					dmas = <&hpdma 75 0x43 0x21 0>;
+					dmas = <&hpdma 75 0x43 0x21>;
 					power-domains = <&CLUSTER_PD>;
 					status = "disabled";
 				};
@@ -1459,7 +1459,7 @@
 					#sound-dai-cells = <0>;
 					clocks = <&rcc CK_KER_SAI2>;
 					clock-names = "sai_ck";
-					dmas = <&hpdma 76 0x43 0x12 0>;
+					dmas = <&hpdma 76 0x43 0x12>;
 					power-domains = <&CLUSTER_PD>;
 					status = "disabled";
 				};
@@ -1485,7 +1485,7 @@
 					#sound-dai-cells = <0>;
 					clocks = <&rcc CK_KER_SAI3>;
 					clock-names = "sai_ck";
-					dmas = <&hpdma 77 0x43 0x21 0>;
+					dmas = <&hpdma 77 0x43 0x21>;
 					power-domains = <&CLUSTER_PD>;
 					status = "disabled";
 				};
@@ -1496,7 +1496,7 @@
 					#sound-dai-cells = <0>;
 					clocks = <&rcc CK_KER_SAI3>;
 					clock-names = "sai_ck";
-					dmas = <&hpdma 78 0x43 0x12 0>;
+					dmas = <&hpdma 78 0x43 0x12>;
 					power-domains = <&CLUSTER_PD>;
 					status = "disabled";
 				};
@@ -1507,8 +1507,8 @@
 				reg = <0x402c0000 0x400>;
 				interrupts-extended = <&exti1 34 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART9>;
-				dmas = <&hpdma 25 0x20 0x10012 0x0>,
-				       <&hpdma 26 0x20 0x3021 0x0>;
+				dmas = <&hpdma 25 0x20 0x10012>,
+				       <&hpdma 26 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART9_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1555,8 +1555,8 @@
 				reg = <0x40330000 0x400>;
 				interrupts-extended = <&exti1 26 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART1>;
-				dmas = <&hpdma 9 0x20 0x10012 0x0>,
-				       <&hpdma 10 0x20 0x3021 0x0>;
+				dmas = <&hpdma 9 0x20 0x10012>,
+				       <&hpdma 10 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_USART1_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1584,7 +1584,7 @@
 					#sound-dai-cells = <0>;
 					clocks = <&rcc CK_KER_SAI4>;
 					clock-names = "sai_ck";
-					dmas = <&hpdma 79 0x63 0x21 0>;
+					dmas = <&hpdma 79 0x63 0x21>;
 					power-domains = <&CLUSTER_PD>;
 					status = "disabled";
 				};
@@ -1595,7 +1595,7 @@
 					#sound-dai-cells = <0>;
 					clocks = <&rcc CK_KER_SAI4>;
 					clock-names = "sai_ck";
-					dmas = <&hpdma 80 0x43 0x12 0>;
+					dmas = <&hpdma 80 0x43 0x12>;
 					power-domains = <&CLUSTER_PD>;
 					status = "disabled";
 				};
@@ -1609,8 +1609,8 @@
 				interrupts = <GIC_SPI 154 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_SPI6>;
 				resets = <&rcc SPI6_R>;
-				dmas = <&hpdma 59 0x20 0x00003012 0>,
-				       <&hpdma 60 0x20 0x00003021 0>;
+				dmas = <&hpdma 59 0x20 0x00003012>,
+				       <&hpdma 60 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI6_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1625,8 +1625,8 @@
 				interrupts = <GIC_SPI 155 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_SPI7>;
 				resets = <&rcc SPI7_R>;
-				dmas = <&hpdma 61 0x20 0x00003012 0>,
-				       <&hpdma 62 0x20 0x00003021 0>;
+				dmas = <&hpdma 61 0x20 0x00003012>,
+				       <&hpdma 62 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI7_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1638,8 +1638,8 @@
 				reg = <0x40370000 0x400>;
 				interrupts-extended = <&exti1 32 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART7>;
-				dmas = <&hpdma 21 0x20 0x10012 0x0>,
-				       <&hpdma 22 0x20 0x3021 0x0>;
+				dmas = <&hpdma 21 0x20 0x10012>,
+				       <&hpdma 22 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART7_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1652,8 +1652,8 @@
 				reg = <0x40380000 0x400>;
 				interrupts-extended = <&exti1 33 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART8>;
-				dmas = <&hpdma 23 0x20 0x10012 0x0>,
-				       <&hpdma 24 0x20 0x3021 0x0>;
+				dmas = <&hpdma 23 0x20 0x10012>,
+				       <&hpdma 24 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART8_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1668,7 +1668,7 @@
 				resets = <&rcc CCI_R>;
 				clocks = <&rcc CK_BUS_CCI>;
 				clock-names = "mclk";
-				dmas = <&hpdma 137 0x60 0x00003012 0>;
+				dmas = <&hpdma 137 0x60 0x00003012>;
 				dma-names = "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_DCMI_PSSI_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1704,7 +1704,7 @@
 					reg = <0x0>;
 					interrupt-parent = <&adc_12>;
 					interrupts = <0>;
-					dmas = <&hpdma 81 0x20 0x12 0x0>;
+					dmas = <&hpdma 81 0x20 0x12>;
 					dma-names = "rx";
 					#io-channel-cells = <1>;
 					#address-cells = <1>;
@@ -1723,7 +1723,7 @@
 					reg = <0x100>;
 					interrupt-parent = <&adc_12>;
 					interrupts = <1>;
-					dmas = <&hpdma 82 0x20 0x12 0>;
+					dmas = <&hpdma 82 0x20 0x12>;
 					dma-names = "rx";
 					#io-channel-cells = <1>;
 					#address-cells = <1>;
@@ -1818,7 +1818,7 @@
 					reg = <0x84 0x70>;
 					#io-channel-cells = <1>;
 					interrupts = <GIC_SPI 184 IRQ_TYPE_LEVEL_HIGH>;
-					dmas = <&hpdma 63 0x63 0x12 0>;
+					dmas = <&hpdma 63 0x63 0x12>;
 					dma-names = "rx";
 					status = "disabled";
 				};
@@ -1828,7 +1828,7 @@
 					reg = <0x104 0x70>;
 					#io-channel-cells = <1>;
 					interrupts = <GIC_SPI 185 IRQ_TYPE_LEVEL_HIGH>;
-					dmas = <&hpdma 64 0x63 0x12 0>;
+					dmas = <&hpdma 64 0x63 0x12>;
 					dma-names = "rx";
 					status = "disabled";
 				};
@@ -1838,7 +1838,7 @@
 					reg = <0x184 0x70>;
 					#io-channel-cells = <1>;
 					interrupts = <GIC_SPI 186 IRQ_TYPE_LEVEL_HIGH>;
-					dmas = <&hpdma 65 0x63 0x12 0>;
+					dmas = <&hpdma 65 0x63 0x12>;
 					dma-names = "rx";
 					status = "disabled";
 				};
@@ -1848,7 +1848,7 @@
 					reg = <0x204 0x70>;
 					#io-channel-cells = <1>;
 					interrupts = <GIC_SPI 187 IRQ_TYPE_LEVEL_HIGH>;
-					dmas = <&hpdma 66 0x63 0x12 0>;
+					dmas = <&hpdma 66 0x63 0x12>;
 					dma-names = "rx";
 					status = "disabled";
 				};
@@ -1858,7 +1858,7 @@
 					reg = <0x284 0x70>;
 					#io-channel-cells = <1>;
 					interrupts = <GIC_SPI 188 IRQ_TYPE_LEVEL_HIGH>;
-					dmas = <&hpdma 67 0x43 0x12 0>;
+					dmas = <&hpdma 67 0x43 0x12>;
 					dma-names = "rx";
 					status = "disabled";
 				};
@@ -1868,7 +1868,7 @@
 					reg = <0x304 0x70>;
 					#io-channel-cells = <1>;
 					interrupts = <GIC_SPI 189 IRQ_TYPE_LEVEL_HIGH>;
-					dmas = <&hpdma 68 0x43 0x12 0>;
+					dmas = <&hpdma 68 0x43 0x12>;
 					dma-names = "rx";
 					status = "disabled";
 				};
@@ -1878,7 +1878,7 @@
 					reg = <0x384 0x70>;
 					#io-channel-cells = <1>;
 					interrupts = <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>;
-					dmas = <&hpdma 69 0x43 0x12 0>;
+					dmas = <&hpdma 69 0x43 0x12>;
 					dma-names = "rx";
 					status = "disabled";
 				};
@@ -1888,7 +1888,7 @@
 					reg = <0x404 0x70>;
 					#io-channel-cells = <1>;
 					interrupts = <GIC_SPI 191 IRQ_TYPE_LEVEL_HIGH>;
-					dmas = <&hpdma 70 0x43 0x12 0>;
+					dmas = <&hpdma 70 0x43 0x12>;
 					dma-names = "rx";
 					status = "disabled";
 				};
@@ -1913,7 +1913,7 @@
 					reg = <0x0>;
 					interrupt-parent = <&adc_3>;
 					interrupts = <0>;
-					dmas = <&hpdma 83 0x20 0x12 0>;
+					dmas = <&hpdma 83 0x20 0x12>;
 					dma-names = "rx";
 					#io-channel-cells = <1>;
 					#address-cells = <1>;
@@ -1946,7 +1946,7 @@
 				interrupts = <GIC_SPI 145 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_HASH>;
 				resets = <&rcc HASH_R>;
-				dmas = <&hpdma 6 0x40 0x3021 0x0>;
+				dmas = <&hpdma 6 0x40 0x3021>;
 				dma-names = "in";
 				feature-domains = <&rifsc STM32MP25_RIFSC_HASH_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -1992,8 +1992,8 @@
 				interrupts = <GIC_SPI 156 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_SPI8>;
 				resets = <&rcc SPI8_R>;
-				dmas = <&hpdma 171 0x20 0x00003012 0>,
-				       <&hpdma 172 0x20 0x00003021 0>;
+				dmas = <&hpdma 171 0x20 0x00003012>,
+				       <&hpdma 172 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_SPI8_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -2005,8 +2005,8 @@
 				reg = <0x46030000 0x400>;
 				interrupts-extended = <&exti2 26 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPUART1>;
-				dmas = <&hpdma 166 0x20 0x10012 0x0>,
-				       <&hpdma 167 0x20 0x3021 0x0>;
+				dmas = <&hpdma 166 0x20 0x10012>,
+				       <&hpdma 167 0x20 0x3021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_LPUART1_ID>;
 				status = "disabled";
@@ -2021,8 +2021,8 @@
 				resets = <&rcc I2C8_R>;
 				#address-cells = <1>;
 				#size-cells = <0>;
-				dmas = <&hpdma 168 0x20 0x00003012 0>,
-				       <&hpdma 169 0x20 0x00003021 0>;
+				dmas = <&hpdma 168 0x20 0x00003012>,
+				       <&hpdma 169 0x20 0x00003021>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_I2C8_ID>;
 				power-domains = <&CLUSTER_PD>;
@@ -2913,9 +2913,9 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				interrupts = <GIC_SPI 122 IRQ_TYPE_LEVEL_HIGH>;
-				dmas = <&hpdma 0 0x62 0x00003101 0x0>,
-				       <&hpdma 0 0x62 0x00003110 0x0>,
-				       <&hpdma 1 0x22 0x00003113 0x0>;
+				dmas = <&hpdma 0 0x62 0x00003101>,
+				       <&hpdma 0 0x62 0x00003110>,
+				       <&hpdma 1 0x22 0x00003113>;
 				dma-names = "tx", "rx", "ecc";
 				status = "disabled";
 			};
--- a/arch/arm64/boot/dts/st/stm32mp25xc.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp25xc.dtsi
@@ -12,8 +12,8 @@
 		interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_CRYP1>;
 		resets = <&rcc CRYP1_R>;
-		dmas = <&hpdma 4 0x40 0x3021 0x0>,
-		       <&hpdma 5 0x43 0x3012 0x0>;
+		dmas = <&hpdma 4 0x40 0x3021>,
+		       <&hpdma 5 0x43 0x3012>;
 		dma-names = "in", "out";
 		feature-domains = <&rifsc STM32MP25_RIFSC_CRYP1_ID>;
 		power-domains = <&CLUSTER_PD>;
@@ -26,8 +26,8 @@
 		interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_CRYP2>;
 		resets = <&rcc CRYP2_R>;
-		dmas = <&hpdma 140 0x40 0x3021 0x0>,
-		       <&hpdma 141 0x43 0x3012 0x0>;
+		dmas = <&hpdma 140 0x40 0x3021>,
+		       <&hpdma 141 0x43 0x3012>;
 		dma-names = "in", "out";
 		feature-domains = <&rifsc STM32MP25_RIFSC_CRYP2_ID>;
 		power-domains = <&CLUSTER_PD>;
--- a/arch/arm64/boot/dts/st/stm32mp25xf.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp25xf.dtsi
@@ -12,8 +12,8 @@
 		interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_CRYP1>;
 		resets = <&rcc CRYP1_R>;
-		dmas = <&hpdma 4 0x40 0x3021 0x0>,
-		       <&hpdma 5 0x43 0x3012 0x0>;
+		dmas = <&hpdma 4 0x40 0x3021>,
+		       <&hpdma 5 0x43 0x3012>;
 		dma-names = "in", "out";
 		feature-domains = <&rifsc STM32MP25_RIFSC_CRYP1_ID>;
 		power-domains = <&CLUSTER_PD>;
@@ -26,8 +26,8 @@
 		interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_CRYP2>;
 		resets = <&rcc CRYP2_R>;
-		dmas = <&hpdma 140 0x40 0x3021 0x0>,
-		       <&hpdma 141 0x43 0x3012 0x0>;
+		dmas = <&hpdma 140 0x40 0x3021>,
+		       <&hpdma 141 0x43 0x3012>;
 		dma-names = "in", "out";
 		feature-domains = <&rifsc STM32MP25_RIFSC_CRYP2_ID>;
 		power-domains = <&CLUSTER_PD>;
